### ISSUE TITLE
Fix checks for nested properties, particularly output-only and settable/updatable

### ISSUE THREAD

**Issue by emilymye** [GoogleCloudPlatform/magic-modules#495]
**tl;dr**: I think the following properties are getting ignored (sometimes) on nested fields, if `properties.select`/`properties.reject` is being used to filter: 
```
:output
:input
:update_url/:update_verb
:url_param_only
:required
:send_empty_value
```
We may want to start a running list of issues and track them using this issue. To document issues I've already found:

#### `output: true`
Really only affects non-Terraform providers (schema checking), but output-only nested fields are often getting sent in requests and are documented as part of valid requests. 

e.g. Ansible: 
https://github.com/GoogleCloudPlatform/magic-modules/blob/63b0c8a48d385973d4ddd95cbbdb7a37690be3f7/templates/ansible/resource.erb#L44-L46

https://github.com/GoogleCloudPlatform/magic-modules/blob/63b0c8a48d385973d4ddd95cbbdb7a37690be3f7/templates/ansible/resource.erb#L111-L113

For example, I have already seen the output-only field compute `diskEncryptionKey.sha256` getting documented/sent as part of requests for Chef/Ansible. It probably works right now because GCP just ignores it but if an API endpoint is stricter (or decides to start being stricter) about request format, this will suddenly become an issue without any MM code changes. 

#### `updatable?` and `settable_properties`/Terraform
Nested properties are less of an issue with Terraform because the schema catches and validates nested properties, but a lot of these resource global vars we set in the template are top-level only:

https://github.com/GoogleCloudPlatform/magic-modules/blob/63b0c8a48d385973d4ddd95cbbdb7a37690be3f7/templates/terraform/resource.erb#L23-L24

Updatable:
https://github.com/GoogleCloudPlatform/magic-modules/blob/63b0c8a48d385973d4ddd95cbbdb7a37690be3f7/provider/terraform.rb#L65-L67

https://github.com/GoogleCloudPlatform/magic-modules/blob/63b0c8a48d385973d4ddd95cbbdb7a37690be3f7/templates/terraform/resource.erb#L35-L37

might cause problems if we have any nested properties that will need to be updated via separate custom update urls (for example, Composer! Composer is fresh in my mind). 

Custom diffs:

https://github.com/GoogleCloudPlatform/magic-modules/blob/63b0c8a48d385973d4ddd95cbbdb7a37690be3f7/templates/terraform/resource.erb#L39-L46

where `settable_properties` (set in the first terraform block shown above) only checks top-level properties for false `output` or `url_param_only`. I think we probably have nested sets, but I *think* we also avoid this problem currently by having custom `diff_suppress_func` overrides for them? 

### How to find/fix? 
I'm not sure what the scope for fixes are, partially because the templates are very complex when it comes to request construction but also because we sometimes just assign empty values or filter out nil/None values in requests and it works out in the end. Unless something is specifically iterating through every property and nested property, it's likely an issue. I briefly tried ctrl-f with `properties.select(&:` or `properties.reject(&:` in .rb/.erb files and you'll see they are often called only on top-level properties.

**Comment by rileykarson** [comment]
@rambleraptor this is the nested prop stuff I was talking about @emilymye working on yesterday

**Comment by rambleraptor** [comment]
I chatted with Emily about the work that she did.

#1340 helps us get to a better place by getting rid of the Nested Object + Array of Nested Object checks that were everywhere. We can now call `.nested_properties` and get a list of nested_properties.

As of now, you could do `.nested_objects.select(&:...)`. This means that you would have the select logic in two places (instead of 4-5 as before). The ideal case would be to have the logic in 1 place, but I don't know how much time we want to dedicate to that.
